src/api-spec/workload-api-spec.yaml

```diff
@@ -0,0 +1,274 @@
+openapi: 3.0.2
+servers:
+  - url: /api/v1
+info:
+  description: "API for the integration of Openshift Telemetry Analytics workload analyis results with any consumer."
+  version: "1.0.0"
+  title: "Openshift Telemetry Workload Analysis API"
+tags:
+  - name: "Accounts"
+    description: "Get the accounts for which data is available."
+  - name: "Workloads"
+    description: "Get workload related details for each account."
+  - name: "Health check"
+    description: "The health checks(liveness and readiness) for this API."
+paths:
+  /:
+    get:
+      tags:
+        - "Health check"
+      summary: "Liveness probe for the service."
+      description: "A probe to check whether the service is up."
+      operationId: "src.routes.health_check.liveness"
+      responses:
+        "200":
+          description: "service is live"
+  /readiness:
+    get:
+      tags:
+        - "Health check"
+      summary: "Check whether the service is ready to serve requests."
+      description: "A readiness probe to check whether the service is ready."
+      operationId: "src.routes.health_check.readiness"
+      responses:
+        "200":
+          description: "service is ready"
+        "500":
+          description: "Service is down/internal exception."
+  /accounts:
+    get:
+      tags:
+        - "Accounts"
+      summary: "Returns a page-by-page list of accounts."
+      description: "Returns a list of accounts. Each account has a one-to-many mapping to Openshift clusters for which data is available with analytics."
+      operationId: "src.routes.accounts.get_accounts"
+      parameters:
+        - in: query
+          name: offset
+          schema:
+            type: integer
+          description: The number of items to skip before starting to collect the result set
+          examples:
+            first:
+              value: 0
+            second:
+              value: 1000
+            third:
+              value: 2000
+        - in: query
+          name: record_count
```

Comment:
We did share it in the initial spec we sent forward, they can't increase this beyond a value but it does allow them to do things like fetch just one record to test API and stuff.

I'll defer to your judgement on whether this should be removed.

Reply:
I dont think it should be exposed. It can be controlled by us.

Reply:
@prashbnair One point of consideration- If we stop letting them maintain the record count, the difference it will create is that they'll have to compute response size every time we send one back. Right now they're in a position where they can limit it as per their script's rate of consumption, as their DB insert too has a latency associated with it.

Reply:
As discussed in the standup, we will continue with exposing record_count as we don't want to move to a `page_number` based approach.